pkg/kubectl/stop.go

```diff
@@ -188,27 +187,9 @@ func (reaper *DaemonSetReaper) Stop(namespace, name string, timeout time.Duratio
 		return "", err
 	}
 
-	// Update the daemon set to select for a non-existent NodeName.
-	// The daemon set controller will then kill all the daemon pods corresponding to daemon set.
-	nodes, err := reaper.Nodes().List(labels.Everything(), fields.Everything())
-	if err != nil {
-		return "", err
-	}
-	var fuzzer = fuzz.New()
-	var nameExists bool
-
-	var nodeName string
-	fuzzer.Fuzz(&nodeName)
-	nameExists = false
-	for _, node := range nodes.Items {
-		nameExists = nameExists || node.Name == nodeName
-	}
-	if nameExists {
-		// Probability of reaching here is extremely low, most likely indicates a programming bug/library error.
-		return "", fmt.Errorf("Name collision generating an unused node name. Please retry this operation.")
+	ds.Spec.Template.Spec.NodeSelector = map[string]string{
```

Comment:
Add a comment what you are doing. ("Update the daemon set to request an unsatisfied selector" or something like that)

Also can you update the documentation, which still says it uses non-existent NodeName:
https://github.com/kubernetes/kubernetes/blob/master/docs/design/daemon.md

```
"delete (if --cascade=true, then first the client turns down all the pods controlled by the DaemonSet (by setting the nodeName to a non-existant name); then it deletes the DaemonSet; then it deletes the pods)"
```